alexandria/core/views.py

```diff
@@ -139,10 +139,32 @@ def multi(self, request, **kwargs):
 
             return response
 
+    @action(methods=["post"], detail=True, url_path="generate-thumbnail")
+    def generate_thumbnail(self, request, pk=None):
+        file = self.get_object()
+        if file.variant == models.File.THUMBNAIL:
+            raise ValidationError(
+                _("File is already a thumbnail, cannot generate thumbnail.")
+            )
+        elif (
+            settings.ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION
+            and not settings.ALEXANDRIA_THUMBNAIL_GENERATE_WITHOUT_HOOK
+        ):
+            raise ValidationError(_("Manual thumbnail generation is not enabled."))
+
+        create_thumbnail(file)
```

Comment:
You should also check if the file already has a thumbnail. This is not done in `create_thumbnail()`, so you may end up with multiple thumbnails if the client calls this endpoint repeatedly